### PR TITLE
Webhook API

### LINKED ISSUE #269: Webhook API
An API to block users.

### PR DESCRIPTION
**Pull request by tsumo**
## PR includes

-   [x] Feature

## Related issues

Resolve #269 

```
$ curl -v -H "X-Api-Key: cba9f9a0-06db-450c-840c-6e83ef9bb4f6" -X GET "http://localhost:3000/api/rest/users?email=user@example.com" 
Note: Unnecessary use of -X or --request, GET is already inferred.
*   Trying 127.0.0.1:3000...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /api/rest/users?email=user@example.com HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/8.1.2
> Accept: */*
> X-Api-Key: cba9f9a0-06db-450c-840c-6e83ef9bb4f6
> 
< HTTP/1.1 200 OK
< set-cookie: next-auth.csrf-token=b43a73268d567ad62904a78f1a721b582a1c7c29897f4f2ba9419e1789158461%7Cd3f66247118d7572d3e631d171d482cf959102c838b7910f07b139318ea0bc76; Path=/; HttpOnly; SameSite=Lax
< set-cookie: next-auth.callback-url=http%3A%2F%2Flocalhost%3A3000; Path=/; HttpOnly; SameSite=Lax
< content-type: application/json
< vary: Accept-Encoding
< date: Tue, 31 Oct 2023 15:02:31 GMT
< connection: close
< transfer-encoding: chunked
< 
* Closing connection 0
{"id":"IRwa8Roc1RIM3-4Ac8aWB","name":"User Name","email":"user@example.com","bonusPoints":0}%
```

### CODE REVIEW

File: src/trpc/trpcBackend.ts [hunk 1]
```diff
@@ -21,7 +24,18 @@ const sessionCheck = t.middleware(({ next, ctx }) => {
     });
 });
 
+const apiTokenCheck = t.middleware(async ({ next, ctx }) => {
+    const header = ctx.headers['x-api-key'];
```
Comment (9teen90nine) [inline src/trpc/trpcBackend.ts:28]: let's use standard authorization header

File: prisma/schema.prisma [hunk 2]
```diff
@@ -188,6 +188,11 @@ model BonusHistory {
   createdAt DateTime @default(dbgenerated("timezone('utc'::text, now())")) @db.Timestamp()
 }
 
+model ApiToken {
+  id          String @id @default(uuid())
```
Comment (9teen90nine) [inline prisma/schema.prisma:192]: I would prefer to use db-generated uids as they  are longer 
`id String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid`

File: src/trpc/trpcBackend.ts (hunk 1 above)
Comment (tsumo) [reply src/trpc/trpcBackend.ts]: fixed

**Review by 9teen90nine — APPROVED** [review]

File: prisma/schema.prisma (hunk 2 above)
Comment (tsumo) [reply prisma/schema.prisma]: fixed
I thought they were the same v4 uuid's

**Review by awinogradov — APPROVED** [review]

**Pull request merged**